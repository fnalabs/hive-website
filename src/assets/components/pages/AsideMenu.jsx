import React from 'react'
import { NavLink } from 'react-router-dom'

import { AnalyticsLink } from 'common'

const AsideMenu = () => (
  <aside className='menu'>
    <p className='menu-label has-text-weight-bold'>v2.1.3</p>
    <ul className='menu-list'>
      <li><NavLink to='/overview' activeClassName='is-active'>Overview</NavLink>
        <ul>
          <li><NavLink to='/model' activeClassName='is-active'>Data Model</NavLink></li>
          <li><NavLink to='/domain' activeClassName='is-active'>Domain Logic</NavLink></li>
          <li><NavLink to='/infrastructure' activeClassName='is-active'>Infrastructure</NavLink></li>
          <li><NavLink to='/telemetry' activeClassName='is-active'>Telemetry</NavLink></li>
        </ul>
      </li>

      <li><NavLink to='/start' activeClassName='is-active'>Get Started</NavLink>
        <ul>
          <li><NavLink to='/setup' activeClassName='is-active'>Setup</NavLink></li>
          <li className='menu-label has-text-weight-bold'>Examples:</li>
          <li><NavLink to='/basic' activeClassName='is-active'>Basic</NavLink></li>
          <li><NavLink to='/rest' activeClassName='is-active'>Rest</NavLink></li>
          <li><NavLink to='/cqrs-es' activeClassName='is-active'>CQRS/ES</NavLink></li>
        </ul>
      </li>

      <li><NavLink to='/documentation' activeClassName='is-active'>Documentation</NavLink>
        <ul>
          <li><NavLink to='/environments' activeClassName='is-active'>Environments</NavLink></li>
          <li><AnalyticsLink to='https://fnalabs.github.io/hive-io/' target='_blank' rel='noopener noreferrer'>API</AnalyticsLink></li>
        </ul>
      </li>
    </ul>
  </aside>
)

export default AsideMenu
