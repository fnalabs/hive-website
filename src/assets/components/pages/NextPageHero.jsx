import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { AnalyticsLink } from 'common'
import { Hero, HeroBody } from 'common/hero'
import { ArrowRight } from 'icons'

const toRegex = /^https:\/\//

export default class NextPageHero extends Component {
  static propTypes = {
    to: PropTypes.string.isRequired
  }

  render () {
    const to = this.props.to

    return (
      <Hero size='small' className='is-hive'>
        <HeroBody centered>
          {toRegex.test(to)
            ? <AnalyticsLink to={to}><ArrowRight height='48' /></AnalyticsLink>
            : <Link to={to}><ArrowRight height='48' /></Link>}
        </HeroBody>
      </Hero>
    )
  }
}
