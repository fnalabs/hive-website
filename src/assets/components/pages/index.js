// site's main pages
export { default as Home } from './home/Home.jsx'
export { default as Overview } from './overview/Overview.jsx'
export { default as Start } from './start/Start.jsx'

// policy pages
export { Cookie } from './policies/Cookie.jsx'
export { Privacy } from './policies/Privacy.jsx'
