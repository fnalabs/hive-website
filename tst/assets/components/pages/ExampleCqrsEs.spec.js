/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { MemoryRouter } from 'react-router'
import ReactGA from 'react-ga'

import { CqrsEs } from 'pages'

jest.mock('react-ga')

describe('<CqrsEs />', () => {
  it('should render CqrsEs page static content', () => {
    const tree = renderer.create(<MemoryRouter><CqrsEs location={{ pathname: '/cqrs-es' }} /></MemoryRouter>).toJSON()

    expect(tree).toMatchSnapshot()
    expect(ReactGA.pageview).not.toBeCalled()
  })

  it('should render CqrsEs page static content and report page view', () => {
    shallow(<CqrsEs location={{ pathname: '/cqrs-es' }} />, {
      context: { isConsent: true }
    })

    expect(ReactGA.pageview).toBeCalledTimes(1)
  })
})
