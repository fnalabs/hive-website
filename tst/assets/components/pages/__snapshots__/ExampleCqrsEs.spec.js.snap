// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CqrsEs /> should render CqrsEs page static content 1`] = `
Array [
  <article
    className="section is-medium"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <header
          className="column is-narrow is-hidden-touch"
        >
          <aside
            className="menu"
          >
            <p
              className="menu-label has-text-weight-bold"
            >
              v2.1.2
            </p>
            <ul
              className="menu-list"
            >
              <li>
                <a
                  aria-current={null}
                  href="/overview"
                  onClick={[Function]}
                >
                  Overview
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/model"
                      onClick={[Function]}
                    >
                      Data Model
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/domain"
                      onClick={[Function]}
                    >
                      Domain Logic
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/infrastructure"
                      onClick={[Function]}
                    >
                      Infrastructure
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/telemetry"
                      onClick={[Function]}
                    >
                      Telemetry
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/start"
                  onClick={[Function]}
                >
                  Get Started
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/setup"
                      onClick={[Function]}
                    >
                      Setup
                    </a>
                  </li>
                  <li
                    className="menu-label has-text-weight-bold"
                  >
                    Examples:
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/basic"
                      onClick={[Function]}
                    >
                      Basic
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/rest"
                      onClick={[Function]}
                    >
                      Rest
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/cqrs-es"
                      onClick={[Function]}
                    >
                      CQRS/ES
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/documentation"
                  onClick={[Function]}
                >
                  Documentation
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/environments"
                      onClick={[Function]}
                    >
                      Environments
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-io/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      API
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </aside>
        </header>
        <section
          className="column content"
          role="document"
        >
          <h1>
            Example: CQRS/ES
          </h1>
          <p>
            This example evolves the previous REST example into a highly distributed architecture in order to handle different magnitudes of network traffic.
          </p>
          <h2
            id="source_code"
          >
            <a
              href="https://www.npmjs.com/package/hive-io-domain-example"
              rel="noopener noreferrer"
              target="_blank"
            >
              Domain Logic
            </a>
             (
            <a
              href="https://github.com/fnalabs/hive-io/tree/master/packages/hive-js-domain-example"
              rel="noopener noreferrer"
              target="_blank"
            >
              Source Code
            </a>
            )
          </h2>
          <div
            className="notification is-info is-light"
          >
            <span
              className="icon"
            >
              <svg
                className="svg-inline"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                  fill="#209cee"
                />
              </svg>
            </span>
            <span>
              You should consider using a private NPM registry or implementing more creative solutions such as extending base Docker images with 
              <code>
                ADD
              </code>
              |
              <code>
                COPY
              </code>
               statements for source code or 
              <code>
                npm link
              </code>
               for your domain logic.
            </span>
          </div>
          <h2
            id="infrastructure"
          >
            <a
              href="https://github.com/fnalabs/hive-io/tree/master/dev/docker/domain/production"
              rel="noopener noreferrer"
              target="_blank"
            >
              Infrastructure
            </a>
          </h2>
          <div
            className="notification is-info is-light"
          >
            <span
              className="icon"
            >
              <svg
                className="svg-inline"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                  fill="#209cee"
                />
              </svg>
            </span>
            <span>
              There is a chicken or egg scenario when you run this example for the first time. In this example, the topics are not created until events are sent from 
              <code>
                hive-producer-js
              </code>
               and 
              <code>
                hive-stream-processor-js
              </code>
              . Therefore, you will need to 
              <strong>
                restart
              </strong>
               
              <code>
                hive-consumer-js
              </code>
               after the topics are created to finally see events flow through the system.
            </span>
          </div>
          <dl>
            <dt>
              <code>
                Producer.dockerfile
              </code>
            </dt>
            <dd>
              <pre>
                FROM fnalabs/hive-producer-js:latest
                <br />
                RUN npm install hive-io-domain-example
              </pre>
            </dd>
            <dt>
              <code>
                Stream-Processor.dockerfile
              </code>
            </dt>
            <dd>
              <pre>
                FROM fnalabs/hive-stream-processor-js:latest
                <br />
                RUN npm install hive-io-domain-example
              </pre>
            </dd>
            <dt>
              <code>
                Consumer.dockerfile
              </code>
            </dt>
            <dd>
              <pre>
                FROM fnalabs/hive-consumer-js:latest
                <br />
                RUN npm install hive-io-domain-example
              </pre>
            </dd>
            <dt>
              <code>
                Rest.dockerfile
              </code>
            </dt>
            <dd>
              <pre>
                FROM fnalabs/hive-base-js:latest
                <br />
                RUN npm install hive-io-domain-example
              </pre>
            </dd>
            <dt>
              <code>
                docker-compose.yml
              </code>
            </dt>
            <dd>
              <pre>
                version: '3.5'
                <br />
                services:
                <br />
                  # proxy for layer 7 routing
                <br />
                  # NOTE: this is an example, you will need to define your own config
                <br />
                  #       ex. https://github.com/fnalabs/hive-io/tree/master/dev/proxy
                <br />
                  proxy:
                <br />
                    image: haproxy:2.3.2-alpine
                <br />
                    container_name: proxy
                <br />
                    depends_on:
                <br />
                      - hive-base-js
                <br />
                      - hive-stream-processor-js
                <br />
                    ports:
                <br />
                      - 80:80
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                    restart: on-failure
                <br />
                <br />
                  # producers
                <br />
                  hive-producer-js:
                <br />
                    build:
                <br />
                      context: .
                <br />
                      dockerfile: Producer.dockerfile
                <br />
                    image: hive-producer-js:production
                <br />
                    container_name: hive-producer-js
                <br />
                    environment:
                <br />
                      ACTOR: ViewContentActor
                <br />
                      ACTOR_LIB: hive-io-domain-example
                <br />
                      ACTOR_URLS: "/contents/:id"
                <br />
                      CLUSTER_SIZE: 1
                <br />
                      HTTP_VERSION: 1
                <br />
                      SECURE: "false"
                <br />
                      TELEMETRY: "true"
                <br />
                      TELEMETRY_URL_METRICS: "http://collector:55681/v1/metrics"
                <br />
                      TELEMETRY_URL_TRACES: "http://collector:55681/v1/trace"
                <br />
                      EVENT_STORE_TOPIC: view
                <br />
                      EVENT_STORE_BROKERS: "kafka:29092"
                <br />
                      EVENT_STORE_ID: producer-client
                <br />
                    depends_on:
                <br />
                      - collector
                <br />
                      - kafka
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                <br />
                  # stream processors
                <br />
                  hive-stream-processor-js:
                <br />
                    build:
                <br />
                      context: .
                <br />
                      dockerfile: Stream-Processor.dockerfile
                <br />
                    image: hive-stream-processor-js:production
                <br />
                    container_name: hive-stream-processor-js
                <br />
                    environment:
                <br />
                      ACTOR: ContentCommandActor
                <br />
                      ACTOR_LIB: hive-io-domain-example
                <br />
                      ACTOR_URLS: "/contents,/contents/:id"
                <br />
                      CLUSTER_SIZE: 1
                <br />
                      HTTP_VERSION: 1
                <br />
                      SECURE: "false"
                <br />
                      TELEMETRY: "true"
                <br />
                      TELEMETRY_URL_METRICS: "http://collector:55681/v1/metrics"
                <br />
                      TELEMETRY_URL_TRACES: "http://collector:55681/v1/trace"
                <br />
                      CACHE_URL: "redis://redis:6379"
                <br />
                      EVENT_STORE_PRODUCER_TOPIC: content
                <br />
                      EVENT_STORE_BROKERS: "kafka:29092"
                <br />
                      EVENT_STORE_ID: stream-processor-client
                <br />
                    depends_on:
                <br />
                      - collector
                <br />
                      - kafka
                <br />
                      - redis
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                  redis:
                <br />
                    image: redis:6.0.9-alpine
                <br />
                    container_name: redis
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                    restart: on-failure
                <br />
                <br />
                  # log stream containers
                <br />
                  kafka:
                <br />
                    image: confluentinc/cp-kafka:5.4.3
                <br />
                    container_name: kafka
                <br />
                    depends_on:
                <br />
                      - zookeeper
                <br />
                    environment:
                <br />
                      KAFKA_ZOOKEEPER_CONNECT: "zookeeper:32181"
                <br />
                      KAFKA_ADVERTISED_LISTENERS: "PLAINTEXT://kafka:29092"
                <br />
                      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
                <br />
                      KAFKA_COMPRESSION_TYPE: gzip
                <br />
                    expose:
                <br />
                      - 29092
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                    restart: on-failure
                <br />
                  zookeeper:
                <br />
                    image: confluentinc/cp-zookeeper:5.4.3
                <br />
                    container_name: zookeeper
                <br />
                    environment:
                <br />
                      ZOOKEEPER_CLIENT_PORT: 32181
                <br />
                    expose:
                <br />
                      - 32181
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                    restart: on-failure
                <br />
                <br />
                  # consumers
                <br />
                  hive-consumer-js:
                <br />
                    build:
                <br />
                      context: .
                <br />
                      dockerfile: Consumer.dockerfile
                <br />
                    image: hive-consumer-js:production
                <br />
                    container_name: hive-consumer-js
                <br />
                    environment:
                <br />
                      ACTOR: ContentEventActor
                <br />
                      ACTOR_LIB: hive-io-domain-example
                <br />
                      CLUSTER_SIZE: 1
                <br />
                      HTTP_VERSION: 1
                <br />
                      SECURE: "false"
                <br />
                      TELEMETRY: "true"
                <br />
                      TELEMETRY_URL_METRICS: "http://collector:55681/v1/metrics"
                <br />
                      TELEMETRY_URL_TRACES: "http://collector:55681/v1/trace"
                <br />
                      EVENT_STORE_TOPIC: "content|view"
                <br />
                      EVENT_STORE_BROKERS: "kafka:29092"
                <br />
                      EVENT_STORE_ID: consumer-client
                <br />
                      EVENT_STORE_GROUP_ID: consumer-group
                <br />
                      EVENT_STORE_FROM_START: "true"
                <br />
                      MONGO_URL: "mongodb://mongo:27017/content"
                <br />
                    depends_on:
                <br />
                      - collector
                <br />
                      - kafka
                <br />
                      - mongo
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                  mongo:
                <br />
                    image: mongo:4.4.2
                <br />
                    container_name: mongo
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                    restart: on-failure
                <br />
                <br />
                  # rest services
                <br />
                  hive-base-js:
                <br />
                    build:
                <br />
                      context: .
                <br />
                      dockerfile: Rest.dockerfile
                <br />
                    image: hive-base-js:production
                <br />
                    container_name: hive-base-js
                <br />
                    environment:
                <br />
                      ACTOR: ContentQueryActor
                <br />
                      ACTOR_LIB: hive-io-domain-example
                <br />
                      ACTOR_URLS: "/contents,/contents/:id"
                <br />
                      CLUSTER_SIZE: 1
                <br />
                      HTTP_VERSION: 1
                <br />
                      SECURE: "false"
                <br />
                      TELEMETRY: "true"
                <br />
                      TELEMETRY_URL_METRICS: "http://collector:55681/v1/metrics"
                <br />
                      TELEMETRY_URL_TRACES: "http://collector:55681/v1/trace"
                <br />
                      MONGO_URL: "mongodb://mongo:27017/content"
                <br />
                    depends_on:
                <br />
                      - collector
                <br />
                      - hive-producer-js
                <br />
                      - mongo
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                <br />
                  # telemetry
                <br />
                  # NOTE: you will need to provide a configuration for the collector
                <br />
                  #       see https://github.com/fnalabs/hive-io/blob/master/dev/collector/collector-config.yml
                <br />
                  collector:
                <br />
                    image: otel/opentelemetry-collector:0.16.0
                <br />
                    container_name: collector
                <br />
                    command: ["--config=/conf/collector-config.yml", "--log-level=ERROR"]
                <br />
                    depends_on:
                <br />
                      - zipkin
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                    restart: on-failure
                <br />
                  zipkin:
                <br />
                    image: openzipkin/zipkin:2.23.1
                <br />
                    container_name: zipkin
                <br />
                    ports:
                <br />
                      - 9411:9411
                <br />
                    networks:
                <br />
                      - hive-io
                <br />
                    restart: on-failure
                <br />
                <br />
                # networking
                <br />
                networks:
                <br />
                  hive-io:
                <br />
                    driver: bridge
              </pre>
            </dd>
          </dl>
        </section>
      </div>
    </div>
  </article>,
  <section
    className="hero is-small is-hive"
  >
    <div
      className="hero-body"
    >
      <nav
        className="level is-mobile"
      >
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="previous page"
            href="/rest"
            onClick={[Function]}
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                />
              </svg>
            </span>
          </a>
        </div>
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="next page"
            href="/documentation"
            onClick={[Function]}
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                />
              </svg>
            </span>
          </a>
        </div>
      </nav>
    </div>
  </section>,
]
`;
